pandas/core/generic.py

```diff
@@ -1408,16 +1408,18 @@ def __neg__(self):
 
     def __pos__(self):
         values = self._values
-        if is_bool_dtype(values):
-            arr = values
-        elif (
-            is_numeric_dtype(values)
+        if (
+            is_bool_dtype(values)
+            or is_numeric_dtype(values)
```

Comment:
The is_numeric_dtype part of this is needed for the fix, the rest is cleanup I figure might as well do while we're here